cou/apps/core.py

```diff
@@ -106,7 +106,44 @@ def post_upgrade_steps(
         :return: List of post upgrade steps.
         :rtype: list[PostUpgradeStep]
         """
-        return self._get_enable_scheduler_step(units) + super().post_upgrade_steps(target, units)
+        return (
+            self._get_enable_scheduler_step(units)
+            + self._get_restart_subordinate_services_steps(units)
+            + super().post_upgrade_steps(target, units)
+        )
+
+    def _get_restart_subordinate_services_steps(
```

Comment:
I think renaming it to `_get_restart_ceilometer_agent_steps` it's cleaner and more accurate

Reply:
If you check the implementation [here](https://review.opendev.org/c/openstack/charm-nova-compute/+/833886), I think `_get_restart_subordinate_services_steps` give us more flexible. I would like to keep this one.

Reply:
The implementation that you shared is `restarting all subordinate services if they aren't running` whereas this is very specific for ceilometer-agent. If we are going to restart all subordinates services like they are doing than I agree with the name.

Reply:
Then let's keep it.